jupyter_server_fileid/manager.py

```diff
@@ -312,21 +312,21 @@ def _from_normalized_path(self, path: Optional[str]) -> Optional[str]:
 
     def _create(self, path: str) -> str:
         path = self._normalize_path(path)
+        row = self.con.execute("SELECT id FROM Files WHERE path = ?", (path,)).fetchone()
+        existing_id = row and row[0]
+
+        if existing_id:
+            return existing_id
+
         id = self._uuid()
         self.con.execute("INSERT INTO Files (id, path) VALUES (?, ?)", (id, path))
         return id
 
     def index(self, path: str) -> str:
+        # create new record
         with self.con:
-            path = self._normalize_path(path)
-            row = self.con.execute("SELECT id FROM Files WHERE path = ?", (path,)).fetchone()
-            existing_id = row and row[0]
-
-            if existing_id:
-                return existing_id
-
-            # create new record
             id = self._create(path)
+            self.con.commit()
```

Comment:
You should be able to remove this since we are using the `Connection context manager here.

Reply:
:+1: